### DISCUSSION TITLE
How to search for a specific data from a JSON file with array in Cypress?

### DISCUSSION THREAD

**Discussion by Deivisom1979** [cypress-io/cypress#21107] [Questions and Help]
Hello everyone!!
I really appreciate if someone help me in my question. 
In this case, the scenario is: Type a TERM# on the #nroTerminal input, click on the Search button and show the results with that TERM#.
I have a JSON file where I mocked it from the API and I need to validate a web page with 3 input fields. 
So, I'm trying to type for a specific terminal# and check if it works and it is not working. Instead of to show only the terminal# on the results, it is showing all the JSON content data. I did the same test validation on the API service and it worked as expected.

Pre Requirement: User should be logged in.

JSON file (attached as TXT file): [carta-cancelamento.txt](https://github.com/cypress-io/cypress/files/8499243/carta-cancelamento.txt)
Screenshot: 
![carta](https://user-images.githubusercontent.com/91226920/163661428-8c2be9b0-d17c-43a9-a553-297d1528fc8f.png)

This is my code in Cypress (search.spec.js)

describe('Search validation', () => {
   cy.intercept(
      'GET',
      '**/mc-extratos/**',
      {fixture: 'financeiro/carta-cancelamento'},
    ).as('verCartas')
   cy.visit('/carta-de-cancelamento')
    cy.get('[data-cy="consultarCartas"]') //button
       .should('be.visible')
       .click()
    cy.wait('@verCartas')
       .its('response.statusCode').should('eq', 200)
    })
    it('Search by Terminal',  () => {
      cy.get('#nroTerminal')
         .should('be.visible')
         .type('05009867')
      cy.get('[data-cy="consultarCartas"]').click()
      cy.wait('@verCartas')
    }
)

**Comment by geromegrignon** [comment]
Hi @Deivisom1979,

If it works as expected by testing your API on its own, it means the data is filtered on API side.
So by mocking the API response, you'll always get the same response whatever the term is: you are mocking the response, not the behavior.

**Reply by Deivisom1979** [reply]
Hi @geromegrignon, 
So, it means that what I want to do/test is not possible if I'm using a JSON file? (Sorry if I misunderstood your explanation, but I'm completely blocked)

**Reply by geromegrignon** [reply]
It's possible by adding a filtering logic in the test.
However it means your are testing the test logic rather than the real one.
If your logic changes, your test will still succeed.

I discourage you to do so.
